src/Presentation/DonationMembershipApplicationAdapter.php

```diff
@@ -51,4 +52,20 @@ private function getMembershipFormBankDataValues( PaymentMethod $paymentMethod )
 			'bankname' => $paymentMethod->getBankData()->getBankName(),
 		];
 	}
+
+	public function getInitialValidationState( Donation $donation ): array {
+		$validationState = [];
+		if ( $donation->getDonor() !== null ) {
+			$validationState['address'] = true;
+		}
+		if ( $donation->getPaymentType() !== PaymentType::DIRECT_DEBIT ) {
+			return $validationState;
+		}
+		/** @var DirectDebitPayment $paymentMethod */
+		$paymentMethod = $donation->getPayment()->getPaymentMethod();
+		if ( $paymentMethod->getBankData()->hasIban() ) {
+			$validationState['bankData'] = true;
+		}
+		return $validationState;
+	}
```

Comment:
Why did you put the code here? I noticed there is no cohesion with the other code in this class and then could not readily come up with a reason to put it here rather than in the route.

Reply:
If the goal was to make testing easier (not needing an e2e test), then how about putting this in a static method in the route? Static is generally not nice though in this case you would not increase the static binding (the only user directly constructs the object), assuming there is no plan to use this in other places (not sure if that even makes sense).

Reply:
> Why did you put the code here? 

I thought of `DonationMembershipApplicationAdapter` as "a class that prepares **all presentation data** for a membership application from a Donation object" rather than "a class that prepares the prepares **only the presentation values** for a membership  application from a Donation object". Everything is conveniently on one place and */* could live with a presentation class that has two methods that do slightly different things.

I don't know what you mean by "putting this in a static method in the route". Do you mean converting the route function into a Handler class, that has a static function that can be tested?

Reply:
Mhh... sure. No objection from my side. Just a bit triggered by lack of cohesion and having an "all the things" thing, even though no real harm seems to be done.

> I don't know what you mean by "putting this in a static method in the route". Do you mean converting the route function into a Handler class, that has a static function that can be tested?

Yeah. Though I was not clued in to the route being a function and not a class at present :)